bugwarrior/config/schema.py

```diff
@@ -172,98 +163,72 @@ def compute_data(cls, values):
     merge_annotations: bool = True
     merge_tags: bool = True
     replace_tags: bool = False
-    static_tags: ConfigList = ConfigList([])
-    static_fields: ConfigList = ConfigList(['priority'])
+    static_tags: ConfigList = []
+    static_fields: ConfigList = ["priority"]
 
-    log_level: typing.Literal[
-        ('DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL', 'DISABLED')
-    ] = 'INFO'
-    log_file: typing.Optional[LoggingPath] = None
+    log_level: Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL", "DISABLED"] = (
+        "INFO"
+    )
+    log_file: typing.Optional[ExpandedPath] = None
 
 
-class Hooks(pydantic.v1.BaseModel):
-    pre_import: ConfigList = ConfigList([])
+class Hooks(BaseConfig):
+    pre_import: ConfigList = []
 
 
-class Notifications(pydantic.v1.BaseModel):
+class Notifications(BaseConfig):
     notifications: bool = False
-    # Although upstream supports it, pydantic has problems with Literal[None].
-    backend: typing.Optional[
-        typing.Literal[('gobject', 'growlnotify', 'applescript')]
-    ] = None
+    backend: typing.Optional[Literal["gobject", "growlnotify", "applescript"]] = None
     finished_querying_sticky: bool = True
     task_crud_sticky: bool = True
     only_on_new_tasks: bool = False
 
 
-class SchemaBase(pydantic.v1.BaseSettings):
-    class Config(PydanticConfig):
-        # Allow extra top-level sections so all targets don't have to be selected.
-        extra = 'ignore'
-
-    hooks: Hooks = Hooks()
-    notifications: Notifications = Notifications()
-
+class SchemaBase(BaseConfig):
+    # Allow extra top-level sections so all targets don't have to be selected.
+    model_config = ConfigDict(extra="ignore")
 
-class ValidationErrorEnhancedMessages(list):
-    """Methods loosely adapted from pydantic.error_wrappers."""
+    hooks: Hooks = Field(default_factory=Hooks)
+    notifications: Notifications = Field(default_factory=Notifications)
```

Comment:
Per [the docs](https://docs.pydantic.dev/latest/concepts/models/#nested-models) it seems like this would work. I'm assuming there's a good reason it doesn't?

```suggestion
    hooks: Hooks
    notifications: Notifications
```

Reply:
In the example you shared, there is no default value because `foo` and `bars` are passed as parameters: 
```
m = Spam(foo={'count': 4}, bars=[{'apple': 'x1'}, {'apple': 'x2'}])

Reply:
You've now reverted back to the way I had it before which still seem like the simplest of the options to me. Thanks.